tools/build/docker/Dockerfile

```diff
@@ -29,25 +38,48 @@ RUN \
     adduser -D -u ${LRR_UID} -G koyomi koyomi; \
 fi
 
+
+# we use s6-overlay-nobin to just pull in the s6-overlay arch agnostic (shell)
+# components, since we apk install the binaries of s6 later which are arch specific
+ADD https://github.com/just-containers/s6-overlay/releases/download/${S6_OVERLAY_RELEASE}/s6-overlay-nobin.tar.gz /tmp/s6-overlay-nobin.tar.gz
+RUN tar -C / -xzf /tmp/s6-overlay-nobin.tar.gz && rm -f /tmp/s6-overlay-nobin.tar.gz
+
+
 WORKDIR /home/koyomi/lanraragi
 
 #Copy cpanfile and install script before copying the entire context
 #This allows for Docker cache to preserve cpan dependencies
-COPY --chown=koyomi:koyomi /tools tools
+COPY --chown=koyomi:koyomi /tools/cpanfile /tools/install.pl tools/
 COPY --chown=koyomi:koyomi /package.json package.json
 
-# Make scripts executable + Run the install script as root
-RUN chmod +x ./tools/build/docker/install-everything.sh ./tools/build/docker/entrypoint.sh && sh ./tools/build/docker/install-everything.sh
+# Install deps (we install Linux::Inotify2 manually since it's not in the base cpanfile as it doesn't build on macOS)
+RUN apk add --update perl perl-io-socket-ssl perl-dev redis libarchive-dev libbz2 openssl-dev zlib-dev \
+imagemagick imagemagick-perlmagick libwebp-tools \
+g++ make pkgconf gnupg wget curl nodejs nodejs-npm \
+shadow s6 s6-portable-utils && \
+curl -L https://cpanmin.us | perl - App::cpanminus && \
+cpanm Linux::Inotify2 && \
+cd tools && cpanm --notest --installdeps . -M https://cpan.metacpan.org && cd .. && \
+npm run lanraragi-installer install-front && \
+apk del perl-dev g++ make gnupg wget curl nodejs nodejs-npm openssl-dev && \
+rm -rf /root/.cpanm/* /root/.npm/ /usr/local/share/man/* node_modules /var/cache/apk/*
 
-#Copy remaining LRR files from context  
+#Copy remaining LRR files from context
+# consider chowning in s6 setup scripts instead
 COPY --chown=koyomi:koyomi /lib lib
 COPY --chown=koyomi:koyomi /public public
 COPY --chown=koyomi:koyomi /script script
 COPY --chown=koyomi:koyomi /templates templates
 COPY --chown=koyomi:koyomi /tests tests
 COPY --chown=koyomi:koyomi /lrr.conf lrr.conf
 COPY --chown=koyomi:koyomi /COPYING COPYING
+COPY /tools/build/docker/redis.conf /etc/redis.conf
+COPY /tools/build/docker/wsl.conf /etc/wsl.conf
```

Comment:
I'm indecisive about the location of `wsl.conf` -- It's technically only for windows builds so it made sense to be in `/build/windows`, but then again it's used during docker builds due to the hybrid nature of the whole thing...

Reply:
Didn't see the .dockerignore comment, guess it's alright that way.